.pre-commit-config.yaml

```diff
@@ -6,3 +6,8 @@ repos:
     hooks:
     - id: flake8
       additional_dependencies: [flake8-docstrings, flake8-debugger, flake8-bugbear]
+- repo: https://github.com/pycqa/isort
+  rev: 5.8.0
+  hooks:
+    - id: isort
+      language_version: python3
```

Comment:
This reformat the files automatically right? I'm not a fan of that. I understand the thought behind it, that all contributed code will be formatted correctly, however in this case only devs using pre-commit will get this.
Having this in the ci-actions might be a better solution to ensure that all new contributions are isorted properly, but as I said, I'm not a fan this, I'd rather do it myself when I edit the file.
And finally, I think this should be taken to a new PR instead and discussed there. If isort is being performed, we're opening the door to other autoformatters: which one should they be, when should they be run, how to we make sure they to not cause havoc in the files they touch?

Reply:
Regarding moving this to another PR: that's fair. I put it here because I thought it was a small change and wasn't as shocking as a full autoformatter.

>  I'm not a fan this, I'd rather do it myself when I edit the file.

So you prefer something to warn you that things are out of order (ex. flake8) rather than just fix them for you?

> how to we make sure they to not cause havoc in the files they touch?

That's what code reviews are for, but also what are you thinking this is going to do? The code is tested so it isn't going to break syntax without us knowing. It doesn't change the actual text of the code beyond whitespace and ordering so it isn't going to change the intended purpose of the code. Even "black" is just moving whitespace around. The only problem I can foresee is if their is a circular import somewhere and the order of imports matters for things to be imported correctly, but even that should be caught by tests and is something we probably don't want to rely on anyway...and if we do run into that we can tell isort to leave the imports how they are with special comments.

> Having this in the ci-actions might be a better solution to ensure

That can be done. Their is a pre-commit bot that can be run and I *think* it will just report success or failure. It can also make a commit to do the updates, but I agree in this case that that causes more confusion/harm especially for new contributors and those unfamiliar with git.

Have you used any auto-formatters? I started to get annoyed by flake8 in the way we use it in pytroll packages because it was so annoying to be told about every little issue that someone else made a long time ago and now I have to add whitespace here or refactor something so newlines don't occur in a certain place. Then I made a PR for pyproj which had black in its pre-commit and it was so nice and easy. "Set it and forget it" as the old infomercials said. It gets to the point where small little preference differences don't matter because it isn't just being checked against some standard it is being fixed and rewritten to that standard for you.

Side note: I just checked pyproj's pre-commit config and apparently there is one for YAML checking. We should get that: https://github.com/pyproj4/pyproj/blob/c714f59c552531cb2834ffc4d81c1e830b4b4043/.pre-commit-config.yaml#L2-L7

I will comment this pre-commit out and move it to another PR.

Reply:
let's discuss in the new PR